### ISSUE TITLE
Surge XT - Filter Cutoff hit / hover region range stretches too far to the right

### ISSUE THREAD

**Issue by VincyZed** [surge-synthesizer/surge#4855]
The hit / hover region for both filter cutoff sliders stretches too far to the right. It takes approximatively half of the freq Offset button on F2, and same for resonance link. When hovering to the left of those buttons and clicking, it's selecting the cutoff silder, so it's not just related to the visual hover.
This may be the case for all horizontal sliders but I'm not sure.

![Hover Bug](https://user-images.githubusercontent.com/50145178/129777834-a8ce40bc-fcaa-43d6-b798-5643c8ebc9ea.gif)

**Comment by mkruselj** [comment]
It's exactly the case for all horizontal sliders. The solution would be placing those two buttons in a higher Z layer so they are on top of the slider. How? I don't know, but JUCE probably has a way.

**Comment by baconpaul** [comment]
oh man that's gonna need a special case. the sliders actually overlap with that button because they are bigger (so you can get the handle all the way over).

@mkruselj is right. We need to zorder them up. they layout in skin order now basically.

**Comment by VincyZed** [comment]
Right, just tested and the range of slider grabbing hasn't changed since 1.9, and is actually good as it is.

**Comment by baconpaul** [comment]
hmm I stuck this at the end of openOrRecreateEditor and it didn't fix it

The problem is these widgets actually are overlapping. I kinda wish they weren't

Ponder ponder

```cpp

    /*
     * OK this is a gross hack but it's OK. The sliders are a bit 'long' so they hover over
     * the switches. Just fix this by brining all the switches to the front. See #4855
     */
    for (auto p : nonmod_param)
    {
        auto s = dynamic_cast<Surge::Widgets::Switch *>(p);
        if (s)
        {
            std::cout << "Got an S" << std::endl;
            s->toFront(false);
        }
    }
```
